addons/lunch/report/lunch_product_report.py

```diff
@@ -53,15 +53,14 @@ def _search_is_available_at(self, operator, value):
 
     @api.multi
     def write(self, values):
-        user_id = self.env.user.id
         if 'is_favorite' in values:
-            for report in self:
-                if values['is_favorite']:
-                    self.env.cr.execute('''INSERT INTO lunch_product_favorite_user_rel(product_id, user_id) VALUES (%s, %s)''',
-                                (report.product_id, user_id))
-                else:
-                    self.env.cr.execute('''DELETE FROM lunch_product_favorite_user_rel WHERE product_id=%s AND user_id=%s''',
-                                (report.product_id, user_id))
+            if values['is_favorite']:
+                commands = [(4, product_id) for product_id in self.mapped('product_id').ids]
+            else:
+                commands = [(3, product_id) for product_id in self.mapped('product_id').ids]
+            self.env.user.write({
```

Comment:
`self.user_id.write()`
modify favorite lunch on the related user for the lunch, and not the current user.

Reply:
Including the fields into the SELF_WRITABLE fields just expose the field to the user for himself, it doesn't prevent modifications by another user.

On the other hand, the ``self.user_id.sudo().write()`` here allows a user to write on any user, so exactly the opposite of what you would like to do.